src/main/java/org/cascadebot/cascadebot/events/VoiceEventListener.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.events;
+
+import net.dv8tion.jda.core.entities.VoiceChannel;
+import net.dv8tion.jda.core.events.guild.voice.GenericGuildVoiceEvent;
+import net.dv8tion.jda.core.events.guild.voice.GuildVoiceJoinEvent;
+import net.dv8tion.jda.core.events.guild.voice.GuildVoiceLeaveEvent;
+import net.dv8tion.jda.core.events.guild.voice.GuildVoiceMoveEvent;
+import net.dv8tion.jda.core.hooks.ListenerAdapter;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.commands.music.SkipCommand;
+import org.cascadebot.cascadebot.data.managers.GuildDataManager;
+import org.cascadebot.cascadebot.utils.votes.VoteButtonGroup;
+
+public class VoiceEventListener extends ListenerAdapter {
+
+    @Override
+    public void onGenericGuildVoice(GenericGuildVoiceEvent event) {
+        // TODO: handle leaving & moving of players so we don't get lonely :(
```

Comment:
This should be completed for this PR

Reply:
Not really as this is more of a beta thing when we allow the bot to stay if people donate